stdlib/n-ary-app.kl

```diff
@@ -109,23 +109,47 @@
                                           (cons-list-syntax 'my-arrow args stx))
                                         stx))])]))]))
 
-(export (rename ([my-arrow ->]) my-arrow))
+(meta
+  (define optional-parens
+    (lambda (symbol)
+      (lambda (stx)
+        (syntax-case stx
+          [(ident x)
+           (pure (list-syntax (symbol) stx))]
+          [(cons x xs)
+           (pure (cons-list-syntax symbol xs stx))])))))
+
+-- TODO: wrap 'datatype' so that parentheses are also optional for
+-- the nullary types and constructors it creates
+(define-macros
+  ([my-Unit (optional-parens 'Unit)]
+   [my-unit (optional-parens 'unit)]
+   [my-Bool (optional-parens 'Bool)]
+   [my-true (optional-parens 'true)]
+   [my-false (optional-parens 'false)]
+   [my-nothing (optional-parens 'nothing)]))
+
+(export (rename ([my-arrow ->]
+                 [my-Unit Unit] [my-unit unit]
+                 [my-Bool Bool] [my-true true] [my-false false]
+                 [my-nothing nothing])
+                my-arrow
```

Comment:
Does this also export `my-arrow` etc? Would we want to use those names in other modules?

Reply:
No, `(rename ([a b]) a)` only exports `b`, it does not also export `a`.